src/marqo/tensor_search/add_docs.py

```diff
@@ -91,7 +91,7 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
                     modality = infer_modality(doc[field])
                     if (modality == Modality.IMAGE and is_no_index) or (
```

Comment:
It's hard to follow the whole logic here. Could it be simplified? Like raise the error earlier if inferred error does not match the field?

Reply:
will refactor in the next release!